### PR TITLE
Infinite scroll: implementation clean up

### PR DESCRIPTION
**Pull request by matyax**
Since we will proabably not need https://github.com/grafana/grafana/pull/81156, I'm bringing here some minor enhancements that I made in that PR.

- Reordered test
- Cleaned up scroll component
- Internally disable by feature flag
- Improved visibility of the lower loader
- **Ensure End of range message is displayed when older logs return no result**

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana/issues/71728

### CODE REVIEW

File: public/app/features/logs/components/InfiniteScroll.tsx [hunk 1]
```diff
@@ -195,27 +210,32 @@ function canScrollTop(
   currentRange: TimeRange,
   timeZone: TimeZone,
   sortOrder: LogsSortOrder
-) {
+): AbsoluteTimeRange | false {
```
Comment (ivanahuckova) [inline public/app/features/logs/components/InfiniteScroll.tsx:213]: nit, but wouldn't `): AbsoluteTimeRange | undefined` be better here. As this function is getter of something, I would expect undefined if it is not able to create it.

File: public/app/features/logs/components/InfiniteScroll.tsx [hunk 2]
```diff
@@ -195,27 +210,32 @@ function canScrollTop(
...
+  return canScroll ? getPrevRange(visibleRange, currentRange) : false;
 }
 
 function canScrollBottom(
   visibleRange: AbsoluteTimeRange,
   currentRange: TimeRange,
   timeZone: TimeZone,
   sortOrder: LogsSortOrder
-) {
+): AbsoluteTimeRange | false {
```
Comment (ivanahuckova) [inline public/app/features/logs/components/InfiniteScroll.tsx:230]: same as above.

**Review by ivanahuckova — COMMENTED** [review]
Awesome that you have added all of the helpful comments. LGTM. Left some nits, but non-blocking.

File: public/app/features/logs/components/InfiniteScroll.tsx (hunk 1 above)
Comment (matyax) [reply public/app/features/logs/components/InfiniteScroll.tsx]: Makes sense! Done.

**Review by ivanahuckova — APPROVED** [review]

**Pull request merged**